schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/dao/BaseCachingDao.java

```diff
@@ -45,7 +55,57 @@ public T get(@Nonnull Key id) throws EntityNotFoundException, IllegalArgumentExc
             return ret;
         }
 
-        return DaoUtil.cachePut(id, meta, super.get(id), Expiration.byDeltaSeconds(cacheExpirySeconds));
+        return DaoUtil.cachePut(id, meta, super.get(id), expiration());
+    }
+
+    @Nonnull
+    @Override
+    public List<T> get(@Nonnull List<Key> ids) throws EntityNotFoundException, IllegalArgumentException {
+        if (!canCache()) {
+            return super.get(ids);
+        }
+
+        List<T> ret = Lists.newArrayList();
+
+        Map<Key, T> cached = DaoUtil.cacheGet(ids, meta);
+
+        List<Key> missing = Lists.newArrayList();
+        for (Key id : ids) {
+            if (cached.containsKey(id)) continue;
+            missing.add(id);
+        }
+
+        if (missing.isEmpty()) {
+            // ALL CACHED
+            for (Key id : ids) {
+                ret.add(cached.get(id));
+            }
+        } else {
+            // Go fetch from datastore
+            List<T> retrieved = super.get(missing);
+
+            //Cache it
+            Map<Key, T> toCache = Maps.newHashMap();
+            for (int i = 0; i < missing.size(); ++i) {
+                toCache.put(missing.get(i), retrieved.get(i));
+            }
+            DaoUtil.cachePutAll(toCache, meta, expiration());
+
+            //Combine previously cached and newly fetched
+            for (Key id : ids) {
+                if (cached.containsKey(id)) {
+                    ret.add(cached.get(id));
+                } else {
+                    ret.add(toCache.get(id));
+                }
+            }
+        }
+
+        return ret;
```

Comment:
Is this showing up here because you merged master into your branch?  Or did you rebase?